payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -418,83 +433,29 @@ internal class PaymentSheetViewModel internal constructor(
     internal class Factory(
         private val applicationSupplier: () -> Application,
         private val starterArgsSupplier: () -> PaymentSheetContract.Args,
+        private val activityLauncherFactory: ActivityLauncherFactory,
+        private val authHostSupplier: () -> AuthActivityStarter.Host,
+        private val eventReporterSupplier: (() -> EventReporter)? = null
     ) : ViewModelProvider.Factory {
         override fun <T : ViewModel?> create(modelClass: Class<T>): T {
-            val application = applicationSupplier()
-            val config = PaymentConfiguration.getInstance(application)
-            val publishableKey = config.publishableKey
-            val stripeAccountId = config.stripeAccountId
-            val stripeRepository = StripeApiRepository(
-                application,
-                { publishableKey }
-            )
-
-            val starterArgs = starterArgsSupplier()
-
-            val googlePayRepository =
-                starterArgs.config?.googlePay?.environment?.let { environment ->
-                    DefaultGooglePayRepository(
-                        application,
-                        environment
-                    )
-                } ?: GooglePayRepository.Disabled
-
-            val prefsRepository = starterArgs.config?.customer?.let { (id) ->
-                DefaultPrefsRepository(
-                    application,
-                    customerId = id,
-                    isGooglePayReady = { googlePayRepository.isReady().first() },
-                    workContext = Dispatchers.IO
+            return DaggerPaymentSheetViewModelComponent.builder()
+                .application(applicationSupplier())
+                .starterArgs(starterArgsSupplier())
+                .eventReporter(
+                    eventReporterSupplier?.let {
```

Comment:
Yeah, it is possible to inject it through `PaymentSheetViewModelModule`, actually it's already [injected](https://github.com/stripe/stripe-android/blob/cf63065e8fc9f23b3fec51e20abce9eb2870e7ba/payments-core/src/main/java/com/stripe/android/paymentsheet/injection/FlowControllerModule.kt#L84) in `FlowControllerModule`.

The reason it's passed as a parameter here is that before the change, two same eventReporter instances are created in [PaymentSheetActivity](https://github.com/stripe/stripe-android/blob/cf63065e8fc9f23b3fec51e20abce9eb2870e7ba/payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetActivity.kt#L79) and [PaymentSheetViewModel](https://github.com/stripe/stripe-android/blob/cf63065e8fc9f23b3fec51e20abce9eb2870e7ba/payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt#L488), since we are not injecting the `PaymentSheetActivity` and an `EventReporter` instance has to be created there, I just passed that instance to the dagger graph. 

We can inject `EventReporter` when some refactoring is done to remove [BaseSheetActivity.eventReporter](https://github.com/stripe/stripe-android/blob/ad426ca33105e7bc5f17c027f8146354d13f75ae/payments-core/src/main/java/com/stripe/android/paymentsheet/ui/BaseSheetActivity.kt#L25)